### PR TITLE
Deleting a profile will delete the membership in the chapter

### PR DESCRIPTION
**Pull request by markd315**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bugfix

* **What is the current behavior?** (You can also link to an open issue here)
Profile.chapter is a dead field with no behavior.
When /profiles/{{profileId}} is DELETEd, the chapter maintains a dead reference to this profile ID in its array.

* **What is the new behavior (if this is a feature change)?**
Profile.chapter is now written (or overwritten) when we add the profile to the chapter.

The chapter deletes the profile from it's member list when /profiles/{{profileId}} is DELETEd.

This is still a one-to-many mapping from chapter to profile for now, each profile can have at most one chapter.

Also, deleting the profile will not delete out-of-date memberships (where the profile does not have its chapter field set), which does apply to some documents currently in the db.

### CODE REVIEW

File: src/domain/chapter/chapter.controller.ts [hunk 1]
```diff
@@ -53,7 +72,7 @@ async function getChapterMembers(req) {
 }
 
 async function getMemberIndex(req) {
-  let members = await chapterRepository.getChapterMembers(req);
+  let members = await chapterRepository.getChapterMembers(req.params.id);
```
Comment (markd315) [inline src/domain/chapter/chapter.controller.ts]: This call was broken to begin with, we were giving it too much info when the repository method just wants a string.

File: src/domain/profile/profile.controller.ts [hunk 2]
```diff
@@ -62,6 +71,16 @@ async function getAllProfiles() {
   return await profileRepository.getAllProfiles()
 }
 
+async function getMemberIndex(chapterId, memberId) {
```
Comment (markd315) [inline src/domain/profile/profile.controller.ts]: yes this is pretty much identical to the method in chaptercontroller.

lmk what to do about it if you care.

File: src/domain/chapter/chapter.controller.ts [hunk 3]
```diff
@@ -32,6 +35,22 @@ async function addChapterMember(req) {
...
+  targetProfile.chapter = chapterId;
+
+  req.body = targetProfile;
+
+  console.log(targetProfile);
+  let updatedProfile = ProfileService.updateProfile(req);
+
+  updatedProfile.id = req.body.id;
+  console.log(updatedProfile);
+
```
Comment (alindenberg) [inline src/domain/chapter/chapter.controller.ts]: Remove console logs

File: src/domain/profile/profile.controller.ts (hunk 2 above)
Comment (alindenberg) [reply src/domain/profile/profile.controller.ts]: All good, we just need these index methods for AWS. Possibly should move to the service but we would need to import the repository into there as well then so it seems counter-productive. All good for now

**Pull request merged**